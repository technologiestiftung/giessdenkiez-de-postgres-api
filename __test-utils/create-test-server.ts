// Based on https://github.com/willianantunes/nextjs-playground/blob/master/__tests__/integration/pages/api/health.spec.test.js
// And https://github.com/Xunnamius/next-test-api-route-handler/blob/main/src/index.ts#L94
// import dns from "dns";
import { VercelRequest, VercelResponse } from "@vercel/node";
import { apiResolver } from "next/dist/server/api-utils/node/api-resolver";
import http, { Server } from "http";
import listen from "async-listen";

/**
 *
 */
export async function createTestServer(
	query: Record<string, string>,
	handler: (
		request: VercelRequest,
		response: VercelResponse
	) => Promise<VercelResponse>
): Promise<{
	server: Server;
	url: string;
}> {
	const server = http.createServer((req, res) => {
		//eslint-disable-next-line @typescript-eslint/no-explicit-any
		return apiResolver(req, res, query, handler, undefined as any, false);
	});
	const url = await listen(server);
	const urlWithParams = `${url}?${new URLSearchParams(query)}`;
	return { server, url: urlWithParams };
}
