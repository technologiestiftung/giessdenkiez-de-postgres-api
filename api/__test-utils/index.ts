import { Generic } from "../_utils/common/interfaces";
import { VercelRequest, VercelResponse } from "@vercel/node";

export function setupRequest(overrides?: Generic): VercelRequest {
	const req = {
		query: {},
		method: "GET",
		...overrides,
	};
	return req as VercelRequest;
}
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function setupResponse(overrides?: Generic): VercelResponse {
	const res = {
		setHeader: jest.fn(),
		query: {},
		...overrides,
	};
	return res as unknown as VercelResponse;
}
