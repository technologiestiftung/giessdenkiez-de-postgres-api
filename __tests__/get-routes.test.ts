import each from "jest-each";
import fetch from "cross-fetch";
import { test, describe, expect } from "@jest/globals";
import v3handler from "../api/v3/get/[type]";
import { createTestServer } from "../__test-utils/create-test-server";
import {
	deleteSupabaseUser,
	truncateTreesAdopted,
	truncateTreesWaterd,
} from "../__test-utils/postgres";
import {
	createSupabaseUser,
	requestSupabaseTestToken,
} from "../__test-utils/req-test-token";

// ██    ██ ██████
// ██    ██      ██
// ██    ██  █████
//  ██  ██       ██
//   ████   ██████

describe("GET v3 routes snapshot tests default responses", () => {
	const email = "foo@example.com";
	const password = "1234567890@";
	beforeAll(async () => {
		await createSupabaseUser(email, password);
	});
	afterAll(async () => {
		await deleteSupabaseUser(email);
	});
	test("should return 200 on wateredbyuser route authenticated", async () => {
		await truncateTreesWaterd();
		const token = await requestSupabaseTestToken(email, password);
		const { server, url } = await createTestServer(
			{ type: "wateredbyuser", uuid: "auth0|abc" },
			v3handler
		);
		// console.log(url);
		const response = await fetch(url, {
			headers: {
				authorization: `Bearer ${token}`,
			},
		});
		server.close();
		const json = await response.json();
		expect(response.status).toBe(200);
		expect(json).toMatchSnapshot();
	});
	test("should return 200 on istreeadopted route authenticated", async () => {
		await truncateTreesWaterd();
		const token = await requestSupabaseTestToken(email, password);
		const { server, url } = await createTestServer(
			{ type: "istreeadopted", id: "_210028b9c8", uuid: "auth0|abc" },
			v3handler
		);
		const response = await fetch(url, {
			headers: {
				authorization: `Bearer ${token}`,
			},
		});
		server.close();
		const json = await response.json();
		expect(response.status).toBe(200);
		expect(json).toMatchSnapshot();
	});

	test("should return 200 on adopted route authenticated", async () => {
		await truncateTreesWaterd();
		const token = await requestSupabaseTestToken(email, password);
		const { server, url } = await createTestServer(
			{ type: "adopted", uuid: "auth0|abc" },
			v3handler
		);
		const response = await fetch(url, {
			headers: {
				authorization: `Bearer ${token}`,
			},
		});
		server.close();
		const json = await response.json();
		expect(response.status).toBe(200);
		expect(json).toMatchSnapshot();
	});
	test("should return 200 on lastwatered route", async () => {
		await truncateTreesWaterd();

		const { server, url } = await createTestServer(
			{ type: "lastwatered", id: "_210028b9c8" },
			v3handler
		);
		const response = await fetch(url);
		server.close();
		const json = await response.json();
		expect(response.status).toBe(200);
		expect(json).toMatchSnapshot();
	});
	test("should return 200 on wateredandadopted route", async () => {
		await truncateTreesWaterd();
		await truncateTreesAdopted();
		const { server, url } = await createTestServer(
			{ type: "wateredandadopted" },
			v3handler
		);
		const response = await fetch(url);
		server.close();
		const json = await response.json();
		expect(response.status).toBe(200);
		expect(json).toMatchSnapshot();
	});

	test("Should return 200 on treesbyid route", async () => {
		const { server, url } = await createTestServer(
			{ type: "treesbyids", tree_ids: "_2100294b1f,_210028b9c8" },
			v3handler
		);
		const response = await fetch(`${url}`);
		server.close();
		const json = await response.json();
		expect(response.status).toBe(200);
		expect(json).toMatchSnapshot();
	});

	test("should return 200 on tree by id route", async () => {
		const { server, url } = await createTestServer(
			{ type: "byid", id: "_2100294b1f" },
			v3handler
		);
		const response = await fetch(`${url}`);
		server.close();
		const json = await response.json();
		expect(response.status).toBe(200);
		expect(json).toMatchSnapshot();
	});

	test("should return 404 on invalid route", async () => {
		const { server, url } = await createTestServer(
			{ type: "invalid" },
			v3handler
		);
		const response = await fetch(`${url}`);
		server.close();
		expect(response.status).toBe(404);
		expect(response.statusText).toBe("Not Found");
		expect(await response.json()).toMatchSnapshot();
	});
});
each([
	[401, "wateredbyuser", { uuid: "123" }, "due to not being authorized"],
	[400, "wateredbyuser", {}, "due to uuid missing", true],

	[400, "istreeadopted", {}, "due to uuid missing", true],
	[400, "istreeadopted", { uuid: "abc" }, "due to id missing", true],
	[
		401,
		"istreeadopted",
		{ uuid: "abc", id: "_21000c10a9" },
		"due to not being authorized",
	],

	[400, "adopted", {}, "due to not uuid missing"],
	[401, "adopted", { uuid: "123" }, "due to not being authorized"],

	[400, "byid", {}, "due to missing id serachParam"],
	[400, "treesbyids", {}, "due to tree_ids missing"],

	[400, "lastwatered", {}, "due to id missing"],
	[
		400,
		"treesbyids",
		{},
		"due to missing tree_ids list serachParam (_2100294b1f,_210028b9c8)",
	],
]).describe(
	"error tests for GET routes",
	(
		statusCode: number,
		type: string,
		overrides: Record<string, unknown>,
		description: string,
		needsAuth?: boolean
	) => {
		const email = "foo@example.com";
		const password = "1234567890@";
		beforeAll(async () => {
			await createSupabaseUser(email, password);
		});
		afterAll(async () => {
			await deleteSupabaseUser(email);
		});
		test(`should return ${statusCode} on route "${type}" ${description}`, async () => {
			const { server, url } = await createTestServer(
				{ type, ...overrides },
				v3handler
			);
			const response = await fetch(`${url}`, {
				headers: {
					...(needsAuth === true && {
						Authorization: `Bearer ${await requestSupabaseTestToken(
							email,
							password
						)}`,
					}),
					"Content-Type": "application/json",
				},
			});
			server.close();
			expect(response.status).toBe(statusCode);
		});
	}
);
