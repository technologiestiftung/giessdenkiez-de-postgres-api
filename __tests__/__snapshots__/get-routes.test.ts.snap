// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET v3 routes snapshot tests default responses Should return 200 on treesbyid route 1`] = `
{
  "data": [
    {
      "adopted": null,
      "art_bot": "Fraxinus ornus",
      "art_dtsch": "Blumen-Esche",
      "baumhoehe": "0",
      "bezirk": "Pankow",
      "caretaker": null,
      "eigentuemer": "Land Berlin",
      "gattung": "FRAXINUS",
      "gattung_deutsch": "ESCHE",
      "geom": {
        "coordinates": [
          13.50326,
          52.64844,
        ],
        "crs": {
          "properties": {
            "name": "EPSG:4326",
          },
          "type": "name",
        },
        "type": "Point",
      },
      "gml_id": "00008100:0028b9c8",
      "hausnr": null,
      "id": "_210028b9c8",
      "kennzeich": "00885",
      "kronedurch": "0",
      "lat": "13.50326",
      "lng": "52.64844",
      "pflanzjahr": 2019,
      "radolan_days": [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        4,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        5,
        6,
        10,
        29,
        37,
        43,
        12,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        16,
        6,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        1,
        2,
        0,
        0,
        0,
        10,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        4,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        8,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
      ],
      "radolan_sum": 205,
      "stammumfg": "0",
      "standalter": "3",
      "standortnr": "5",
      "strname": null,
      "type": null,
      "watered": null,
      "zusatz": null,
    },
    {
      "adopted": null,
      "art_bot": "Carpinus betulus",
      "art_dtsch": "Hainbuche",
      "baumhoehe": "7",
      "bezirk": "Pankow",
      "caretaker": null,
      "eigentuemer": "Land Berlin",
      "gattung": "CARPINUS",
      "gattung_deutsch": "HAINBUCHE",
      "geom": {
        "coordinates": [
          13.50295,
          52.64778,
        ],
        "crs": {
          "properties": {
            "name": "EPSG:4326",
          },
          "type": "name",
        },
        "type": "Point",
      },
      "gml_id": "00008100:00294b1f",
      "hausnr": null,
      "id": "_2100294b1f",
      "kennzeich": "41374",
      "kronedurch": "2",
      "lat": "13.50295",
      "lng": "52.64778",
      "pflanzjahr": 2019,
      "radolan_days": [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        4,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        5,
        6,
        10,
        29,
        37,
        43,
        12,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        16,
        6,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        1,
        2,
        0,
        0,
        0,
        10,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        4,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        8,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
      ],
      "radolan_sum": 205,
      "stammumfg": "31",
      "standalter": "3",
      "standortnr": "74/2",
      "strname": "Hörstenweg",
      "type": null,
      "watered": null,
      "zusatz": null,
    },
  ],
  "error": null,
  "links": {},
  "name": "@technologiestiftung/giessdenkiez-de-postgres-api",
  "range": {
    "end": 1,
    "start": 0,
    "total": 2,
  },
  "url": "/?type=treesbyids&tree_ids=_2100294b1f%2C_210028b9c8",
}
`;

exports[`GET v3 routes snapshot tests default responses should return 200 on adopted route authenticated 1`] = `
{
  "data": [],
  "error": null,
  "links": {},
  "name": "@technologiestiftung/giessdenkiez-de-postgres-api",
  "range": {
    "end": -1,
    "start": -1,
    "total": 0,
  },
  "url": "/?type=adopted&uuid=auth0%7Cabc",
}
`;

exports[`GET v3 routes snapshot tests default responses should return 200 on istreeadopted route authenticated 1`] = `
{
  "data": false,
  "error": null,
  "name": "@technologiestiftung/giessdenkiez-de-postgres-api",
  "url": "/?type=istreeadopted&id=_210028b9c8&uuid=auth0%7Cabc",
}
`;

exports[`GET v3 routes snapshot tests default responses should return 200 on lastwatered route 1`] = `
{
  "data": [],
  "error": null,
  "links": {},
  "name": "@technologiestiftung/giessdenkiez-de-postgres-api",
  "range": {
    "end": -1,
    "start": -1,
    "total": 0,
  },
  "url": "/?type=lastwatered&id=_210028b9c8",
}
`;

exports[`GET v3 routes snapshot tests default responses should return 200 on tree by id route 1`] = `
{
  "data": [
    {
      "adopted": null,
      "art_bot": "Carpinus betulus",
      "art_dtsch": "Hainbuche",
      "baumhoehe": "7",
      "bezirk": "Pankow",
      "caretaker": null,
      "eigentuemer": "Land Berlin",
      "gattung": "CARPINUS",
      "gattung_deutsch": "HAINBUCHE",
      "geom": {
        "coordinates": [
          13.50295,
          52.64778,
        ],
        "crs": {
          "properties": {
            "name": "EPSG:4326",
          },
          "type": "name",
        },
        "type": "Point",
      },
      "gml_id": "00008100:00294b1f",
      "hausnr": null,
      "id": "_2100294b1f",
      "kennzeich": "41374",
      "kronedurch": "2",
      "lat": "13.50295",
      "lng": "52.64778",
      "pflanzjahr": 2019,
      "radolan_days": [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        4,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        5,
        6,
        10,
        29,
        37,
        43,
        12,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        3,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        16,
        6,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        1,
        2,
        0,
        0,
        0,
        10,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        4,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        8,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
      ],
      "radolan_sum": 205,
      "stammumfg": "31",
      "standalter": "3",
      "standortnr": "74/2",
      "strname": "Hörstenweg",
      "type": null,
      "watered": null,
      "zusatz": null,
    },
  ],
  "error": null,
  "name": "@technologiestiftung/giessdenkiez-de-postgres-api",
  "url": "/?type=byid&id=_2100294b1f",
}
`;

exports[`GET v3 routes snapshot tests default responses should return 200 on wateredandadopted route 1`] = `
{
  "data": [],
  "error": null,
  "links": {},
  "name": "@technologiestiftung/giessdenkiez-de-postgres-api",
  "range": {
    "end": -1,
    "start": -1,
    "total": 0,
  },
  "url": "/?type=wateredandadopted",
}
`;

exports[`GET v3 routes snapshot tests default responses should return 200 on wateredbyuser route authenticated 1`] = `
{
  "data": [],
  "error": null,
  "links": {},
  "name": "@technologiestiftung/giessdenkiez-de-postgres-api",
  "range": {
    "end": -1,
    "start": -1,
    "total": 0,
  },
  "url": "/?type=wateredbyuser&uuid=auth0%7Cabc",
}
`;

exports[`GET v3 routes snapshot tests default responses should return 404 on invalid route 1`] = `
{
  "error": "invalid route invalid",
}
`;
